### ISSUE TITLE
[Enhancement] Allow Disposal Tracks to be redstone controlled

### ISSUE THREAD

**Issue by wshadow** [CovertJaguar/Railcraft#647]
The wiki says "The Disposal Track is basically a hybrid between the Disembarking Track and the Suspended Track. " [1]

I think it would be great if the disposal track could be redstone controlled, kind of like the Disembarking Track can. But maybe in reverse (i.e. a redstone signal disables the disembarking part) to:
a) not break existing builds
b) less headache for people who just want to use it without building fancy contraptions.

Having this would allow my villager trading hall to be a lot less bulky ;)

[1] http://railcraft.info/wiki/track:disposal_track

**Comment by Forecaster** [comment]
How would you deliver a redstone signal to the track without putting a block next to it, potentially preventing it from working?

**Comment by wshadow** [comment]
Well, that is where you have to get creative when you want to use that feature. But recently I've had no problems dropping villagers in a 1x1 hole in the floor, so I think it should be quite doable.

**Comment by Vexatos** [comment]
Making the Disposal Track react to redstone would mean it will have to contain Redstone in its recipe (like all the other tracks that react to redstone). This would also mean that it would probably have to me made out of advanced rail (like all the other tracks that react to redstone). Even if not, one would have to change the texture so it contains redstone and if advanced rails are included it would not even match the suspended tracks anymore, making it quite ugly when suspended.

**Comment by wshadow** [comment]
1. Cost:
  * to make a Disposal Track you already need steel plates, which means at least a blast furnace, a rollling machine and a hobbyist steam engine.
  * a piece of redstone or 1-2 advanced rails should be quite doable at that point.
2. Redstone in the recipe for consistency:
  * I hadn't thought of that yet, but I agree that you should need redstone somewhere in there.
  * Some ideas to include that: replace a wooden tie with redstone or use a 'steel plate mechanism' instead of just a 'steel plate'. ('steel plate mechanism' would be a new item containing at least some redstone and a steel plate)
  * there is a lot of grey in the texture. I think I could add some red pixels to hint at the redstone interaction. If someone else has a better idea for that -> let me know. I'm not the best with regards to graphics.
3. Advanced rails:
  * I'm not sure if those are needed for consistency. The redstone is meant to _disable_ the steel (plate) trap door (and only that), like a lock preventing it from opening. 
  * even if we include advanced rails in the recipe, we still have different options for the texture (maybe do sth. like the Control Track recipe+texture)
4. Texture:
  * already covered in the last 2 points.
  * to summarize: we have a lot of grey pixels to play with and can reuse concepts like the Control Track
